cloudify_kubernetes/tasks.py

```diff
@@ -158,6 +158,60 @@ def _do_resource_status_check(resource_kind, response):
         else:
             ctx.logger.debug('status {0}'.format(status))
 
+    elif resource_kind == 'Deployment':
+        conditions = response['status']['conditions']
+        if conditions:
+            for condition in conditions:
+                if condition['type'] == 'Available':
+                    ctx.logger.debug('Deployment condition is Available')
+
+                elif condition['type'] == 'ReplicaFailure':
+                    raise NonRecoverableError(
+                        'Deployment condition is ReplicaFailure ,'
+                        'reason:{0}, message: {1}'
+                        ''.format(condition['reason'], condition['message']))
+
+                elif condition['type'] == 'Progressing':
+                    raise OperationRetry(
+                        'Deployment condition is Progressing')
+        else:
+            raise OperationRetry('Deployment condition is not ready yet')
+
+    elif resource_kind == 'PersistentVolumeClaim':
+        status = response['status']['phase']
+        if status in ['Pending', 'Available']:
```

Comment:
I think Available should be logged, not-retry. Thoughts?

Reply:
Well, The initial status for PersistentVolumeClaim will be Available and then it will even transition to  Bound or Pending based if it is found the suitable PersistentVolume available  for the request

Reply:
What if it is not to be used in the same deployment?